test/tests/deepkitType.test.ts

```diff
@@ -0,0 +1,95 @@
+import { classToPlain, t, plainToClass } from '@deepkit/type';
+import { getModelForClass, prop } from '../../src/typegoose';
+
+enum Group {
+  confidential = 'confidential',
+  public = 'public',
+}
+
+class Account {
+  @t.mongoId.group(Group.public)
+  public _id: string;
+
+  @t.group(Group.public)
+  public __v: number;
+
+  @t.group(Group.public)
+  @prop()
+  public email: string;
+
+  @t.group(Group.confidential)
+  @prop()
+  public confidentialProperty?: string;
+}
+
+const AccountModel = getModelForClass(Account);
+
+describe('@deepkit/type transforms', () => {
+  let id: string;
+  let accountClassObject: Account;
+  let account: Account;
+  beforeEach(async () => {
+    accountClassObject = new Account();
+    accountClassObject.email = 'somebody@gmail.com';
+    accountClassObject.confidentialProperty = 'secret';
+    account = await AccountModel.create(accountClassObject);
+    // note here that _id is an ObjectId, hence the toString()
+    // otherwise it will have the shape of : { _bsonType: 'ObjectId', id: ArrayBuffer }
```

Comment:
can this comment be removed? (is this residue from copying the class-transformer tests?)

Reply:
Sure, it can be removed. Should I do it or will you?

Scott

Reply:
you can if you want, if not i will do it later before merge (when other stuff is resolved)

Reply:
@hasezoey - I think my lack of Git experience is showing. I've got 8 conflicting files now when pulling from my origin and I'm not sure how to resolve them properly. Just force the changes? I don't want to mess any work you did either. So, if you could do the changes, I'd appreciate it.

Reply:
i actually added a comment to show how to update your local branch after the rebase i had done: https://github.com/typegoose/typegoose/pull/602#issuecomment-911700895

TL;DR: run `git pull -f` (this will replace the local branch with the changes on the remote by force, meaning it will remove all local changes that do not align with the remote branch)

Reply:
fixed in 29a667f4aff4dcc8d84bca59c516a72b4de796de